sample/dynamic.go

```diff
@@ -94,5 +90,22 @@ func (d *DynamicSampler) GetSampleRate(trace *types.Trace) (rate uint, keep bool
 			d.Metrics.Gauge(name, val)
 		}
 	}
-	return rate, shouldKeep, "dynamic", key
+	return rate, "dynamic", key
+}
+
+func (d *DynamicSampler) MakeSamplingDecision(rate uint, trace *types.Trace) bool {
```

Comment:
I am pretty happy with the removal of the keep parameter, but I thought I understood that we were not going to add a new function to every sampler. I think MakeSamplingDecision is the same for all samplers, except for the metrics. Why is this better than centralizing that into the call from collect()?

Reply:
Unfortunately, the `DeterministicSampler`'s decision making logic is quite different from the rest of the samplers

Reply:
Ugh. You're right, but deterministic sampling breaks with the throughput throttle anyway, so maybe we should consider treating it differently or exempting it from the throttle.